k8s/cmd/nsmdp/nsmdp.go

```diff
@@ -231,6 +226,78 @@ func NewNSMDeviceServer(serviceRegistry serviceregistry.ServiceRegistry) error {
 	return err
 }
 
+func (nsm *nsmClientEndpoints) addClientDevice() {
+	nsm.mutext.Lock()
+	defer nsm.mutext.Unlock()
+
+	for {
+		nsm.clientId++
+
+		id := fmt.Sprintf("nsm-%d", nsm.clientId)
+		if _, ok := nsm.devs[id]; ok {
+			// Item already exists, increment to new one.
+			continue
+		}
+		// Add a new device Id
+		dev := &pluginapi.Device{
+			ID:     id,
+			Health: pluginapi.Healthy,
+		}
+		nsm.devs[id] = dev
+		nsm.resp.Devices = append(nsm.resp.Devices, dev)
+		break
+	}
+
+}
+
+func (n *nsmClientEndpoints) sendDeviceUpdate() {
+	n.mutext.Lock()
+	defer n.mutext.Unlock()
+	if n.pluginApi != nil {
+		api := *n.pluginApi
+		logrus.Infof("Send device plugins intfo %v", n.resp)
+		if err := api.Send(n.resp); err != nil {
+			logrus.Errorf("Failed to send response to kubelet: %v\n", err)
+		}
+	}
+}
+
+func (n *nsmClientEndpoints) reciveWorkspaces() {
```

Comment:
s/recive/receive

PS.  And probably something a bit more cosmetic, but acronyms like API, NSM or NSMD should be all uppercase based on Go's naming convention. There's a couple of places that you can update to resolve that both here and in the PR before this one 👍

Reply:
Thanks, nsm here is name of internal structure, so I could not make it uppercase.

Reply:
@haiodo I was referring to the case when the acronym is in the middle of the namings, for ex. `deployNsmdAndDataplane` should be `deployNSMDAndDataplane`